src/service/data_products/taxa_count.py

```diff
@@ -232,11 +255,49 @@ async def get_taxa_counts(
     q = await _query(store, collection_id, load_ver, rank)
     for dp_proc in [dp_match, dp_sel]:
         await _add_subset_data_in_place(q, store, collection_id, load_ver, rank, dp_proc)
-        # For now always sort by the std data. See if ppl want sort by match/selection data
-        # before implementing something more sophisticated.
+
+    _sort_taxa_counts(q, sort_priority, [dp_match, dp_sel])
+
     return _taxa_counts(dp_match=dp_match, dp_sel=dp_sel, data=q)
 
 
+def _fill_missing_orders(sort_order: list[str]):
+    # fill in missing orders with the default sort order
+    if not isinstance(sort_order, list):
+        raise ValueError(f"sort_order must be a list of strings, provided: {sort_order}")
+
+    return [order for order in _DEFAULT_SORT_ORDER if order not in sort_order] + sort_order
+
+
+def _sort_taxa_counts(
+        q: list[dict[str, Any]],
+        sort_priority: list[str],
+        dp_list: list[models.DataProductProcess]):
+    # Sort taxa count records in place by the sort_priority list.
+    processed_count = [_TYPE2FIELD[dp.type] for dp in dp_list if dp]
+
+    if processed_count:
+
+        if len(sort_priority) != len(set(sort_priority)):
+            raise errors.IllegalParameterError(f"Duplicate sort priority found: {sort_priority}")
+
+        try:
+            sort_order_rev = [_SORT_PRIORITY_ORDER_MAP[k] for k in sort_priority]
```

Comment:
```suggestion
            sort_order_rev = [_SORT_PRIORITY_ORDER_MAP[k] for k in sort_priority
                                         if _SORT_PRIORITY_ORDER_MAP[k] in processed_count]
```
Or we could throw an error if the user asks for a sort where the order includes a match or selection they didn't provide

Reply:
I think we want to set sort_priority to "selected, matched, standard" by default. That way, we shouldn't raise if the process hasn't started.

Reply:
👍 set sort_priority to None by default and raise error if sort_priority has count that's not processed.

Reply:
:+1: